### PR TITLE
Image mms pinned messages

### PR DESCRIPTION
**Pull request by veken1199**
PR for story: #20

### CODE REVIEW

**Review by g-harel — DISMISSED** [review]
I need to fix the ui tests since there is a conflict with master

File: src/org/thoughtcrime/securesms/ConversationFragment.java [hunk 1]
```diff
@@ -326,17 +326,24 @@ public int compare(MessageRecord lhs, MessageRecord rhs) {
...
     pinHandler       = handler;
     databaseToQuery  = pinHandler.getAppropriateDatabase(message);
 
-    if(!message.isMms()) {
-      if (pin) {
+    if (pin) {
+      // Blocking Video and Audio pinning Temporarily
+        if(message.isMms()) {
+          MediaMmsMessageRecord mediaMessage = (MediaMmsMessageRecord) message;
+          if (!mediaMessage.getSlideDeck().getThumbnailSlide().getContentType().contains("image")) {
```
Comment (shelbaz) [inline src/org/thoughtcrime/securesms/ConversationFragment.java:339]: Is GIF considered image? Does it pin successfully? 
As well, does documents crash it too?

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java [hunk 2]
```diff
@@ -18,124 +18,124 @@
...
+                handler.handleUnpinMessage(record, handler.getAppropriateDatabase(record));
+                ((ViewGroup) v.getParent().getParent().getParent()).removeAllViews();
             }
         });
     }
 
     private void setMessageView(MessageRecord record, ViewHolder viewHolder) {
         if (record.isOutgoing()) {
             viewHolder.recipient.setText(R.string.PinnedMessageActivity_own_name);
-            return;
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: eh why you remove this ?

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java [hunk 3]
```diff
@@ -18,124 +18,124 @@
...
         }
 
         String messageSenderName = record.getRecipient().getName();
-
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: you kill mi format

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java [hunk 4]
```diff
@@ -25,86 +26,85 @@
...
 
 import static android.widget.RelativeLayout.ALIGN_PARENT_LEFT;
 import static android.widget.RelativeLayout.ALIGN_PARENT_RIGHT;
 
 public class PinnedMessageAdapter extends RecyclerView.Adapter<PinnedMessageAdapter.ViewHolder> {
-    private Cursor               dataCursor;
     private Context              context;
-    private MmsSmsDatabase       db;
+    private Cursor               dataCursor;
+    private GlideRequests  glideRequests;
```
Comment (erinb90) [inline src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: formatting

Comment (veken1199) [reply src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: @erinb90 yes I noticed that

File: src/org/thoughtcrime/securesms/ConversationFragment.java (hunk 1 above)
Comment (veken1199) [reply src/org/thoughtcrime/securesms/ConversationFragment.java:339]: @shelbaz yes, it pinning Gif works, But the document is still breaking

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java [hunk 5]
```diff
@@ -17,6 +17,7 @@
 package org.thoughtcrime.securesms;
 
 import android.app.Activity;
+
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: extra space intentional?

File: src/org/thoughtcrime/securesms/ConversationItem.java [hunk 6]
```diff
@@ -602,9 +602,19 @@ public void onClick(View v, final Slide slide) {
...
+
+    public ThumbnailClickListener(MessageRecord record) {
+        conversationRecipient = record.getRecipient();
+        messageRecord         = record;
+        pin                   = true;
+    }
+
     public void onClick(final View v, final Slide slide) {
-      if (shouldInterceptClicks(messageRecord) || !batchSelected.isEmpty()) {
+      if (shouldInterceptClicks(messageRecord) || !batchSelected.isEmpty() && !pin) {
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/ConversationItem.java]: `this.pin` would be nice.

File: src/org/thoughtcrime/securesms/PinnedMessageAdapter.java [hunk 7]
```diff
@@ -136,22 +137,33 @@ private void setMessageView(MessageRecord record, ViewHolder viewHolder) {
 
             viewHolder.recipient.setText(R.string.PinnedMessageActivity_own_name);
             return;
+
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/PinnedMessageAdapter.java]: .

File: src/org/thoughtcrime/securesms/PinnedMessageHandler.java [hunk 8]
```diff
@@ -40,4 +40,5 @@ public boolean handleUnpinMessage(final MessageRecord message, MessagingDatabase
         }
         return result;
     }
+
```
Comment (g-harel) [inline src/org/thoughtcrime/securesms/PinnedMessageHandler.java]: .

**Comment by erinb90** [comment]
New location of pin indicator:
screenshot_2018-03-03-18-47-26

**Comment by veken1199** [comment]
Looks good, can be merged

**Review by g-harel — APPROVED** [review]

**Pull request merged**